lib/Scheduler/sched_region.cpp

```diff
@@ -49,6 +49,18 @@ SchedRegion::SchedRegion(MachineModel *machMdl, DataDepGraph *dataDepGraph,
   schedUprBound_ = INVALID_VALUE;
 
   spillCostFunc_ = spillCostFunc;
+
+  Config &schedIni = SchedulerOptions::getInstance();
+  DumpDDGs_ = schedIni.GetBool("DUMP_DDGS", false);
+  DDGDumpPath_ = schedIni.GetString("DDG_DUMP_PATH", "");
+
+  if (DumpDDGs_) {
+    if (DDGDumpPath_.empty())
+      llvm::report_fatal_error(
+          "DDG_DUMP_PATH must be set if trying to DUMP_DDGS.", false);
```

Comment:
I prefer a hard error. If I miss-set the path, I prefer an immediate exit because then I can immediately tell that something is wrong, rather than wait for the run to finish to tell that something is wrong. I specifically did this for empty() to force the user of `DUMP_DDGS` to also uncomment the `DDG_DUMP_PATH` line in the sched.ini and set a value, to make it easier to use correctly.